### ISSUE TITLE
How to train LLama-70B eagle?

### ISSUE THREAD

**Issue by YixinSong-e** [SafeAILab/EAGLE#38]
I have a finetuned llama-70B model, but I can't run this project correctly due to OOM. I have 8 80G-A100.

**Comment by Liyuhui-12** [comment]
What training configuration did you use? We trained using 4xA100 40G. Below is our training configuration.

train_config={
    "lr":3e-5,
    "bs":4,
    "gradient_accumulation_steps":1,
    "datapath":f"{args.tmpdir}",
    "is_warmup":True,
    "num_epochs":200,
    "num_warmup_steps":2000,
    "total_steps":800000,
    "p_w":0.1,
    "v_w":1.0,
    "head_w":0.1,
    "num_workers":2,
    "embeding":True,
    "act":"No",
    "data_noise":True,
    "noise":"uniform",
    "mean":0.0,
    "std":0.2,
    "residual":"true,norm",
    "max_len":1200,
    "config_path":"config.json",
    "b1":0.9,
    "b2": 0.95,
    "grad_clip": 0.5,
}

**Comment by xiongqisong** [comment]
I face the same error when train BlueLM-7B-Chat，use A100 4 GPU，80G memory each GPU，OOM happened when run accelerator.backward(), below is the result of torch.cuda.memory_summary():

**Comment by xiongqisong** [comment]
|===========================================================================|
|                  PyTorch CUDA memory summary, device ID 0                 |
|---------------------------------------------------------------------------|
|            CUDA OOMs: 0            |        cudaMalloc retries: 0         |
|===========================================================================|
|        Metric         | Cur Usage  | Peak Usage | Tot Alloc  | Tot Freed  |
|---------------------------------------------------------------------------|
| Allocated memory      |  57135 MiB |  66653 MiB | 133472 MiB |  76336 MiB |
|       from large pool |  57070 MiB |  66588 MiB | 133367 MiB |  76297 MiB |
|       from small pool |     65 MiB |     66 MiB |    104 MiB |     39 MiB |
|---------------------------------------------------------------------------|
| Active memory         |  57135 MiB |  69270 MiB | 133472 MiB |  76336 MiB |
|       from large pool |  57070 MiB |  69205 MiB | 133367 MiB |  76297 MiB |
|       from small pool |     65 MiB |     66 MiB |    104 MiB |     39 MiB |
|---------------------------------------------------------------------------|
| Requested memory      |  57132 MiB |  69269 MiB | 133398 MiB |  76266 MiB |
|       from large pool |  57067 MiB |  69204 MiB | 133293 MiB |  76226 MiB |
|       from small pool |     65 MiB |     66 MiB |    104 MiB |     39 MiB |
|---------------------------------------------------------------------------|
| GPU reserved memory   |  63572 MiB |  74920 MiB |  74920 MiB |  11348 MiB |
|       from large pool |  63506 MiB |  74852 MiB |  74852 MiB |  11346 MiB |
|       from small pool |     66 MiB |     68 MiB |     68 MiB |      2 MiB |
|---------------------------------------------------------------------------|
| Non-releasable memory |   6436 MiB |   7324 MiB |  58828 MiB |  52391 MiB |
|       from large pool |   6435 MiB |   7323 MiB |  58707 MiB |  52271 MiB |
|       from small pool |      0 MiB |      2 MiB |    120 MiB |    119 MiB |
|---------------------------------------------------------------------------|
| Allocations           |     443    |     665    |    2461    |    2018    |
|       from large pool |     274    |     496    |    1954    |    1680    |
|       from small pool |     169    |     171    |     507    |     338    |
|---------------------------------------------------------------------------|
| Active allocs         |     443    |     665    |    2461    |    2018    |
|       from large pool |     274    |     496    |    1954    |    1680    |
|       from small pool |     169    |     171    |     507    |     338    |
|---------------------------------------------------------------------------|
| GPU reserved segments |     285    |     331    |     331    |      46    |
|       from large pool |     252    |     297    |     297    |      45    |
|       from small pool |      33    |      34    |      34    |       1    |
|---------------------------------------------------------------------------|
| Non-releasable allocs |      61    |      95    |    1116    |    1055    |
|       from large pool |      58    |      94    |     948    |     890    |
|       from small pool |       3    |       3    |     168    |     165    |
|---------------------------------------------------------------------------|
| Oversize allocations  |       0    |       0    |       0    |       0    |
|---------------------------------------------------------------------------|
| Oversize GPU segments |       0    |       0    |       0    |       0    |
|===========================================================================|

**Comment by Liyuhui-12** [comment]
I suspect it might be due to the following reasons:

- In the json file pointed to by *--configpath*, *num_hidden_layers* is not set to 1.
- The *max_len* in the *train_config* is not correctly set.

**Comment by xiongqisong** [comment]
3Q for the answer，nice! Train process is running~^_^
